Spanner/src/Connection/Grpc.php

```diff
@@ -1387,9 +1387,10 @@ private function createTransactionSelector(array &$args)
                 $transaction['begin'] = $this->formatTransactionOptions($transaction['begin']);
             }
 
-            $selector = $this->serializer->decodeMessage($selector, $transaction);
+            $str = json_encode($transaction);
```

Comment:
The `json_encode` method accepts the [JSON_FORCE_OBJECT](https://www.php.net/manual/en/json.constants.php#constant.json-force-object) flag, which will ensure that an empty array `[]` will be encoded in JSON as `{}`

```suggestion
            $str = json_encode($transaction, JSON_FORCE_OBJECT);
```